model/jpa/src/main/java/org/keycloak/events/jpa/JpaAdminEventQuery.java

```diff
@@ -153,11 +153,11 @@ public AdminEventQuery maxResults(int maxResults) {
 
         TypedQuery<AdminEventEntity> query = em.createQuery(cq);
 
-        if (firstResult != null) {
+        if (firstResult != null && firstResult > 0) {
```

Comment:
I'd say it doesn't provide big benefit because we still need to check and set `maxResults` to `DEFAULT_MAX_RESULTS`. See the else clause for details.

Reply:
I added workaround for HHH-14295 to `paginateQuery` functionso it should be safe to use here. See: https://github.com/keycloak/keycloak/pull/7673/commits/623eb2f925816a0bfb14431974403170702cec08#diff-1f7f50ee362cfb68b3f5ff709fc15ab52205fe78a1190642b7ca1db284817185R13-R16